### PR TITLE
Allow embedding txiki in other applications

### PR DESCRIPTION
**Pull request by jspngh**
Hi,

First of all, thanks for all the work on txiki.
Currently only a command-line interface is available, but I would like to embed this javascript runtime in my own application.

Is this change something that you would be open to?
It requires only a small adaptation of the CMakeLists.txt and Makefile.

Let me know what you think.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -152,13 +151,22 @@ string(TOLOWER ${CMAKE_SYSTEM_NAME} TJS_PLATFORM)
...
 target_include_directories(tjs PRIVATE ${CURL_INCLUDE_DIRS})
+target_include_directories(tjs PUBLIC src)
 target_link_libraries(tjs qjs uv_a m3 sqlite3 m pthread ${CURL_LIBRARIES})
 
 if (BUILD_WITH_MIMALLOC)
     target_compile_definitions(tjs PRIVATE TJS__HAS_MIMALLOC)
     target_link_libraries(tjs mimalloc-static)
 endif()
 
+add_executable(tjs-cli EXCLUDE_FROM_ALL
```
Comment (saghul) [inline CMakeLists.txt:162]: Don't exclude it, it needs to always be built.

File: Makefile [hunk 2]
```diff
@@ -22,7 +22,7 @@ $(BUILD_DIR):
 	cmake -B $(BUILD_DIR) -DCMAKE_BUILD_TYPE=$(BUILDTYPE)
 
 $(TJS): $(BUILD_DIR)
-	cmake --build $(BUILD_DIR) -j $(JOBS)
+	cmake --build $(BUILD_DIR) --target tjs-cli -j $(JOBS)
```
Comment (saghul) [inline Makefile:25]: You don't need this if you don't exclude it.

File: CMakeLists.txt (hunk 1 above)
Comment (saghul) [reply CMakeLists.txt:162]: If you'd like to only build the library, then add some BUILD_ONLY_LIBRARY option please.

File: CMakeLists.txt [hunk 3]
```diff
@@ -152,13 +151,22 @@ string(TOLOWER ${CMAKE_SYSTEM_NAME} TJS_PLATFORM)
...
+    src/cli.c
+)
+target_link_libraries(tjs-cli tjs)
```
Comment (saghul) [inline CMakeLists.txt:165]: Don't we need all the linked libraries here instead?

**Review by saghul — CHANGES_REQUESTED** [review]
Alright, let's do this! Left some comments.

File: CMakeLists.txt (hunk 3 above)
Comment (jspngh) [reply CMakeLists.txt:165]: For me, it works without specifying all the libraries.
I think we can either:
- leave it as it is now
- add the `PUBLIC` scope for the `target_link_libraries` of the tjs library
- add all the libraries here again

What do you prefer?

Comment (saghul) [reply CMakeLists.txt:165]: Let's leave it for now then.

**Comment by jspngh** [comment]
> Alright, let's do this! Left some comments.

Should be fixed now, let me know if there's anything else.

**Review by saghul — APPROVED** [review]

**Pull request merged**